src/main/java/cubicchunks/world/column/CubeMap.java

```diff
@@ -208,6 +204,6 @@ private int binarySearch(int cubeY) {
 			}
 		}
 
-		return mid; // not found :(
+		return start; // not found :(
```

Comment:
That is why it will not work, I needed a method that returns where a value should be, even if it does not exist... It makes it a bit more complex to use (see contains()) but its worth it.
The tests I just created should catch any bugs if we change tweak things in the future.